haystack/nodes/preprocessor/splitter.py

```diff
@@ -0,0 +1,502 @@
+            self.tokenizer = FeatureExtractor(pretrained_model_name_or_path=tokenizer_model)
+
+        if split_by == "token" and not self.tokenizer:
+            raise ValueError(
+                "If you set split_by='token', you must give a value to 'tokenizer_model'. "
+                "Use the same model you're using for your Reader."
+            )
+
+    def _validate_split_params(
+        self, split_by: SplitBy, split_regex: Optional[str], split_length: int, split_overlap: int
+    ):
+        """
+        Performs some basic validation on the parameters of the splitter.
+        """
+        if split_by not in get_args(SplitBy):
+            raise ValueError(f"split_by must be one of: {', '.join(get_args(SplitBy))}")
+
+        if not isinstance(split_length, int) or split_length <= 0:
+            raise ValueError("split_length must be an integer > 0")
+
+        if split_length:
+            if not isinstance(split_overlap, int) or split_overlap < 0:
+                raise ValueError("split_overlap must be an integer >= 0")
+
+            if split_overlap >= split_length:
+                raise ValueError("split_length must be higher than split_overlap")
+
+        if split_regex and not split_by == "regex":
+            logger.warning(
+                "You provided a value to 'split_regex', but 'split_by=\"%s\"'. "
+                "The document will be split by %s and the regex pattern will be ignored.",
+                split_by,
+                split_by,
+            )
+
+    @property
+    def nltk_language(self):
+        return self._nltk_language
+
+    @nltk_language.setter
+    def nltk_language(self, nltk_language):
+        self.sentence_tokenizer = load_sentence_tokenizer(
+            language=nltk_language, tokenizer_model_folder=self.nltk_folder
+        )
+        self._nltk_language = nltk_language
+
+    @property
+    def nltk_folder(self):
+        return self._nltk_folder
+
+    @nltk_folder.setter
+    def nltk_folder(self, nltk_folder):
+        self.sentence_tokenizer = load_sentence_tokenizer(
+            language=self.nltk_language, tokenizer_model_folder=nltk_folder
+        )
+        self._nltk_folder = nltk_folder
+
+    def run(  # type: ignore
+        self,
+        documents: List[Document],
+        split_by: Optional[SplitBy] = None,
+        split_regex: Optional[str] = None,
+        split_length: Optional[int] = None,
+        split_overlap: Optional[int] = None,
+        split_max_chars: Optional[int] = None,
+        add_page_number: Optional[bool] = None,
+    ):
+        """
+        Perform document splitting on a document. This method can split on different units, at different lengths,
+        and include some overlap across the splits. It can also properly assign page numbers and re-assign headlines
+        found in the metadata to each split document.
+
+        No char should be lost in splitting, not even whitespace, and all headlines should be preserved.
```

Comment:
Can we say "is" instead of "Should"? Should sounds like we're not sure it's really going to happen :)

Reply:
With PreProcessor we never know :joy: Fixing